compose/config/config.py

```diff
@@ -421,17 +428,31 @@ def env_vars_from_file(filename):
     return env
 
 
-def resolve_volume_paths(volumes, working_dir=None):
+def resolve_volume_paths(service_dict, working_dir=None):
     if working_dir is None:
         raise Exception("No working_dir passed to resolve_volume_paths()")
 
-    return [resolve_volume_path(v, working_dir) for v in volumes]
+    return [
+        resolve_volume_path(v, working_dir, service_dict['name'])
+        for v in service_dict['volumes']
+    ]
 
 
-def resolve_volume_path(volume, working_dir):
+def resolve_volume_path(volume, working_dir, service_name):
     container_path, host_path = split_path_mapping(volume)
     container_path = os.path.expanduser(container_path)
+
     if host_path is not None:
+        if not any(host_path.startswith(c) for c in PATH_START_CHARS):
```

Comment:
https://docs.python.org/2/library/stdtypes.html#str.startswith

Pretty minor, `startswith()` also accepts a tuple, so this could be

```python
if not host_path.startswith(PATH_START_CHARS):
    ...
```

(if `PATH_START_CHARS` is changed to tuple)